### PR TITLE
Reworks BoH bombs

### PR DESCRIPTION
**Pull request by zxaber**
## About The Pull Request
Changes BoH bombs to only affect non-anchored items directly, aside from ripping up floor tiles. This means the tear will no longer destroy the hull or heavily destroy departments. It *will* still kill anyone that's not buckled down to something (and always kills the user), and loose items like tools will fly across the room, potentially breaking stuff.

https://streamable.com/r5g3w

To compensate a bit, the rift now stays active for eight seconds, up from five.

Behind the scenes, the rift is no longer a child of singularities, as the number of exceptions needed were starting to stack up.

Also adds an admin message for every person that falls into the rift, to help with tallying up the deaths should such be required.

## Why It's Good For The Game
We have enough bomb items that space departments and require a bunch of time to patch over. This keeps the BoH bombing as a still-deadly tool for anyone with a death objective (or antags that just want to kill people) without being an easy mid-round department destroying device (or shuttle call forcer).

## Changelog
:cl:
balance: BoH bombs reworked: The rift will no longer pull anchored objects, or anyone bound to such, and will no longer destroy the station's hull. Loose items, floor tiles, and employees not holding onto something remain at risk.
/:cl:

An additional video perhaps better showing the change;
https://streamable.com/6p8yl

I'm not 100% certain that SSfastprocess is ideal, as I'm not sure how susceptible this will be to lagging servers late-game. But SSobj felt super underwhelming.

### CODE REVIEW

**Comment by TheVekter** [comment]
This kind of kills the entire idea of BoH bombs. I really see no reason to change this. If anything, this is a buff to the station against cult since BoH can kill Nar'sie.

**Comment by AdamElTablawy** [comment]
Come on.

We removed it spawning a singularity. Fine. One person being able to instantly end the round was a meme.

We removed the permanent rift to hell. Fine. Creating an impassable, unfixable hole in reality was probably also much.

But now it won't even damage the station? What's the point if it kills you and destroys your body, then? At that point I'd rather set off a suicide grenade or a maxcap.

**Comment by Fikou** [comment]
with every year my friend bohbomb gets more shit and shit

**Comment by carshalash** [comment]
Does this mean no more "heehoo funny 10 BoH in a locker" bombs?

**Comment by ATH1909** [comment]
https://youtu.be/F2ohslGivm8

**Comment by Pumpkinoe** [comment]
well now most grenades are better
what's even the lore reason behind it? it just... "oh no my floor tiles" now?
honestly you should just make it a little more concentrated on a single area and i'd be happy

File: code/datums/components/storage/concrete/bag_of_holding.dm [hunk 1]
```diff
@@ -16,8 +16,59 @@
...
+			if(move == src || move.anchored)
+				continue
+			if(ismob(move))
+				var/mob/M = A
+				if(M.mob_negates_gravity())
+					continue
+			move.safe_throw_at(T, 5, 1, force = MOVE_FORCE_EXTREMELY_STRONG)
+
+	for(var/atom/movable/target in T.contents)
+		if(target == src || target.anchored || istype(target, /mob/dead/observer))
```
Comment (XDTM) [inline code/datums/components/storage/concrete/bag_of_holding.dm]: will this delete camera mobs?

**Comment by interestingusernam3** [comment]
ided

**Comment by daklj** [comment]
still not hugbox enough 
how about it just kills the person who used it and removes 1 floor tile?

**Comment by AdmiralPancakes1** [comment]
come on at least make it rip up the plating

**Comment by HomocidlePancakes** [comment]
*Nerfs noone asked for: Part 542*

I’ll just say the same thing I said the last time someone tried to gut it - give it a channel and pop some bolded text in chat. People with 2 BoHs walking around are super fucking obvious. Shit, you can make them give out a glow and flavor it as “bluespess interference” or whatever.

**Comment by SpinnerMaster** [comment]
mmm this water is a bit too spicy for me. Very bad nerf to BOHBomb, might as well remove the functionality entirely. I can make a better bomb with gibbonite and some scrap parts from maint.

**Comment by Pumpkinoe** [comment]
If you REALLY want to nerf it for some reason
Just make it have some sort of delay so people can run maybe?

**Comment by zxaber** [comment]
To be clear, I didn't want to nerf the killing capacity, I wanted to nerf the station damaging capacity. That's why it didn't destroy hull or uproot anchored objects but still chewed through anyone nearby.

I called it a "rework" because I had basically re-coded the rift to not be a child of singularities. I suppose "refactor" would have been more appropriate.

**Pull request closed without merging**